app/services/call_event.rb

```diff
@@ -0,0 +1,47 @@
+module CallEvent
+  class Base
+    def self.publish(call)
+      new(call).publish
+    end
+
+    def initialize(call)
+      @call = call
+    end
+
+    def publish
+      ChampaignQueue.push(payload)
```

Comment:
I'm going to try and figure out another way to pass the URL params. My initial approach was to create an action, but then I realized it didn't make sense. If I create an action for every call, then I also need to update the action every time the call gets updated. And if we ask ourselves why are we creating an action in the first place, there's no good reason, other than to comply old code that's hard to change and extend.